x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/nlp/FillMaskProcessor.java

```diff
@@ -79,10 +80,11 @@ static InferenceResults processResult(
         String resultsField
     ) {
         if (tokenization.getTokenizations().isEmpty() || tokenization.getTokenizations().get(0).getTokens().length == 0) {
-            return new WarningInferenceResults("No valid tokens for inference");
+            throw new ElasticsearchStatusException("tokenization is empty", RestStatus.INTERNAL_SERVER_ERROR);
```

Comment:
Isn't this a bad request exception? If yes, we should reuse `ExceptionsHelpers.badRequestException`. Same stands for all the other exceptions in the PR that are marked as internal server error.